templates/fine-tune-llama2/train.py

```diff
@@ -0,0 +1,871 @@
+    config.update(
+        **{
+            "as_test": args.as_test,
+            "batch_size": BS,
+            "batch_size_per_device": args.batch_size_per_device,
+            "block_size": args.ctx_len,
+            "ds_config": CONFIG_DIR,
+            "eval_batch_size": args.eval_batch_size_per_device,
+            "grad_accum": args.grad_accum,
+            "lora": args.lora,
+            "lr": LR,
+            "mixed_precision": args.mixed_precision,
+            "model_name": MODEL_ID,
+            "no_grad_ckpt": args.no_grad_ckpt,
+            "num_checkpoints_to_keep": args.num_checkpoints_to_keep,
+            "num_devices": ND,
+            "num_epochs": args.num_epochs,
+            "output_dir": args.output_dir,
+            "seed": 42,
+            "special_token_path": args.special_token_path,
+            "stop_perplexity": args.stop_perplexity,
+            "train_path": args.train_path,
+            "test_path": args.test_path,
+        }
+    )
+
+    # Add LoRA config if needed
+    if config["lora"]:
+        with open("./lora_configs/lora.json", "r") as json_file:
+            lora_config = json.load(json_file)
+        config["lora_config"] = lora_config
+
+    # Add deepspeed plugin to the config
+    ds_plugin = DeepSpeedPlugin(hf_ds_config=config.get("ds_config"))
+    config.update(ds_plugin=ds_plugin)
+
+    # Set the local checkpoint directory which may get used when the remote cloud storage bucket checkpointing is disabled.
+    os.environ["RAY_AIR_LOCAL_CACHE_DIR"] = config["output_dir"]
+
+    # Set up runtime_env for Ray Train
+    ray.init(
+        runtime_env={
+            "env_vars": {
+                "HF_HOME": "/mnt/local_storage/.cache/huggingface",
+                "RAY_AIR_LOCAL_CACHE_DIR": os.environ["RAY_AIR_LOCAL_CACHE_DIR"],
+            },
+            # This is not needed to run in Workspace. However, it might be needed for other environments.
+            # "working_dir": ".",
+        }
+    )
+
+    # Read data
+    train_ds = ray.data.read_json(config["train_path"])
+    if args.test_path is not None:
+        valid_ds = ray.data.read_json(config["test_path"])
+    else:
+        valid_ds = None
+
+    # Read special tokens json file
+    with open(config["special_token_path"], "r") as json_file:
+        special_tokens = json.load(json_file)["tokens"]
+
+    # Specify the cloud bucket path to use when saving training artifacts
+    # ANYSCALE_ARTIFACT_STORAGE is the default cloud storage bucket created for Anyscale users
+    artifact_storage = os.environ.get("ANYSCALE_ARTIFACT_STORAGE", "artifact_storage")
+    user_name = re.sub(r"\s+", "__", os.environ.get("ANYSCALE_USERNAME", "user"))
+    model_name=config.get("model_name")
+    storage_path = (
+        f"{artifact_storage}/{user_name}/ft_llms_with_deepspeed/{model_name}"
+    )
+
+    # trial_name = f"{model_name}".split("/")[-1]
+    # if config["lora"]:
+    #     trial_name += "-lora"
```

Comment:
This can also go, right?

Reply:
It seems so. I'm not very sure but:
-  it's not used in other places.
-  it works fine without it

let's remove for now